### ISSUE TITLE
[WP6.4] Rename UI is displayed in the advanced panel of all blocks

### ISSUE THREAD

**Issue by t-hamano** [WordPress/gutenberg#55953]
### Description

In the latest Gutenberg, rename UI is enabled for almost all blocks, but as of WordPress 6.4, [only group blocks should be renameable](https://wordpress.org/documentation/wordpress-version/version-6-4/#rename-group-blocks). However, except for group blocks, there is a rename UI in the Advanced panel for almost all blocks.

If you enter a new block name in this panel, it will not be reflected in the list view. It also doesn't seem to be saved as attributes.

The rename menu is not displayed in the list view or block toolbar menu unless it is a group block.

### Step-by-step reproduction instructions

- Inserts some block other than the group block.
- Open the Advanced panel and verify that the "Block Name" field is present.
- The block name in the list view should not change when you enter text.

### Screenshots, screen recording, code snippet

https://github.com/WordPress/gutenberg/assets/54422211/1e6f7b63-cfad-4380-aff6-7cc937cfaf36

### Environment info

WordPress 6.4 (General Release Version)

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by Mamaduka** [comment]
This was introduced during refactoring in #55604. The control uses `hasBlockSupport( name, 'renaming', true );` check, which doesn't ship with WP 6.4.

cc @getdave

**Comment by getdave** [comment]
That's a bug if it's happenning for 6.4. Renaming support for all blocks should be in Gutenberg `trunk` only.

**Comment by getdave** [comment]
Fix for this will need to land (I assume) in `6.4.1`. 

```diff
diff --git a/packages/block-editor/src/hooks/block-rename.js b/packages/block-editor/src/hooks/block-rename.js
index d8f414a3e9..60d13760f2 100644
--- a/packages/block-editor/src/hooks/block-rename.js
+++ b/packages/block-editor/src/hooks/block-rename.js
@@ -16,7 +16,11 @@ export const withBlockRenameControl = createHigherOrderComponent(
 	( BlockEdit ) => ( props ) => {
 		const { name, attributes, setAttributes, isSelected } = props;
 
-		const supportsBlockNaming = hasBlockSupport( name, 'renaming', true );
+		const supportsBlockNaming = hasBlockSupport(
+			name,
+			'__experimentalMetadata',
+			false
+		);
 
 		return (
 			<>
```

Pinging @mikachan for advice on next steps.

This happened as the result of changes in `trunk` becoming out of sync with 6.4.

**Comment by mikachan** [comment]
Thanks for the ping! I believe the next steps to include a fix for 6.4.1 would be to create a PR against the `wp/6.4` branch with the above change and add the `Backport to WP Minor Release` label.

**Comment by getdave** [comment]
Thanks. Working on a PR now.

**Comment by MadtownLems** [comment]
It looks like this missed 6.4.1.  Can someone confirm it's on the list for 6.4.2? A new field that does nothing (and has no explanatory text) is pretty confusing 😅

**Comment by getdave** [comment]
@mikachan Did we not land [the fix](https://github.com/WordPress/gutenberg/pull/55995) in 6.4.1?

**Comment by mikachan** [comment]
This fix didn't go out with 6.4.1 as that was an emergency release for a single fix on the Core side. This fix will be going out with 6.4.2, alongside some other GB package updates: https://github.com/WordPress/wordpress-develop/pull/5665

**Comment by t-hamano** [comment]
Update: WordPress 6.4.2 was released on December 6th, but this issue does not seem to be resolved. My understanding is that 6.4.2 is a maintenance security release. As stated in [this core ticket](https://core.trac.wordpress.org/ticket/59828), it will be fixed when 6.4.3 is released.

**Comment by mikachan** [comment]
> My understanding is that 6.4.2 is a maintenance security release. As stated in [this core ticket](https://core.trac.wordpress.org/ticket/59828), it will be fixed when 6.4.3 is released.

This is my understanding, too 👍

**Comment by t-hamano** [comment]
Now that WordPress 6.4.3 has been [released](https://wordpress.org/news/2024/01/wordpress-6-4-3-maintenance-and-security-release/), I tested this issue again. This issue appears to have been fixed, so I would like to close it. Thank you everyone!
